app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationDataTransformer.kt

```diff
@@ -0,0 +1,353 @@
+
+        return VoterRegistration(
+            registrationHelper = buildRegistrationHelper(),
+            dateOfBirth = Dates.formatAsISO8601_ShortDate(newRegistrationData.birthday), //TODO DateAdapter used short date, is this correct?
+            mailingAddress = mailingAddress?.toApiAddressData(),
+            previousRegistrationAddress = addressData.previousAddress?.toApiAddressData(),
+            registrationAddress = addressData.homeAddress.toApiAddressData(),
+            registrationAddressIsMailingAddress = !addressData.isMailingAddressDifferent,
+            name = newRegistrationData.name.toApiName(),
+            previousName = newRegistrationData.previousName?.toApiName(),
+            gender = Gender.fromPrefix(newRegistrationData.name.title).toString(),
+            race = race, // TODO race is required in the API, but not in the app?
+            party = party,
+            voterClassifications = buildVoterClassifications(),
+            signature = Signature(
+                mimeType = "image/png",
+                image = signature
+            ),
+            voterIds = buildVoterIds(),
+            contactMethods = buildContactMethods(),
+            additionalInfo = buildAdditionalInfo()
+        )
+
+    }
+
+    private fun buildRegistrationHelper(): RegistrationHelper? {
+        if (!assistanceData.hasSomeoneAssisted) return null // TODO is this correct, or should this be based on the canvasser and always non-null?
+
+        val helperPhone = assistanceData.helperPhone ?: throw InvalidRegistrationException(
+            "helperPhone is null",
+            R.string.assistants_phone_number_is_empty
+        )
+
+        return RegistrationHelper(
+            registrationHelperType = "assistant", // TODO is this a default value?
+            name = assistanceData.helperName?.toApiName(),
+            address = assistanceData.helperAddress?.toApiAddressData(),
+            contactMethods = listOf(
+                ContactMethod(
+                    type = "phone",
+                    value = helperPhone,
+                    capabilities = listOf("voice") // TODO should capabilties be more here? IIRC, this is only VOICE from [ApiContactMethod]
+                )
+            )
+        )
+    }
+
+    private fun buildVoterClassifications(): List<VoterClassification> {
+        val is18OnElectionDay = VoterClassification(
+            type = "eighteen_on_election_day",
+            assertion = newRegistrationData.is18OrOlderByNextElection
+        )
+
+        val isUsCitizen = VoterClassification(
+            type = "united_states_citizen",
+            assertion = newRegistrationData.isUsCitizen
+        )
+
+        val sendCopyInEmail = VoterClassification(
+            type = "send_copy_in_mail", // TODO Is this "I give permission to OSET Org to send me news and updates?" It's looking like it may not be, that might be "opt_in_email"? Where does this come from?
```

Comment:
no, that's opt_in_email

I don't believe "send_copy_in_mail" used in the app. It was probably planned but never used. we can remove it.

Reply:
1. Ok, yeah, looks like this wasn't used in the previous version. Sorry, I hadn't realized how much was removed from the original, and now that I'm looking at it, it's all a bit more clear.

2. I think we have a discrepancy on `opt_in_email` so just making sure here... the setter for `opt_in_email` in the original `RockyRequest` was not in use, and was always set to `false` in the `ApiRockyRequest` builder. In this case, that checkbox appears to have been setting `partner_opt_in_email` instead:
```
        subscriptions.add(RxCompoundButton.checkedChanges(emailOptIn)
                .observeOn(Schedulers.io())
                .debounce(DEBOUNCE, TimeUnit.MILLISECONDS)
                .skip(1)
                .subscribe(checked -> db.update(
                        RockyRequest.TABLE,
                        new RockyRequest.Builder()
                                .partnerOptInEmail(checked)
                                .build(),
                        RockyRequest._ID + " = ? ", String.valueOf(rockyRequestRowId.get()))));
```
Does that sound correct?